### PR TITLE
Support folder level foundation

### LINKED ISSUE #40: Add folder foundation recipe

### LINKED ISSUE #49: Support folder level audit as a component
The in the foundation recipe to choose org or folder level

### LINKED ISSUE #329: Unable to create a nested folder for prod
Hi,

I am trying to create an environment by following full.hcl example.
I need to create the prod folder under an existing folder structure, so I added `parent_type` and `parent_id` to prod folder data element as follows. I would like to ensure that all subsequents services project for data/app/firebase are created under **prod/app-team**

      # Top level prod folder.
      template "folder_prod" {
        recipe_path = "{{$recipes}}/folder.hcl"
        output_path = "./live"
        data = {
          display_name = "prod"
          parent_type = "folder"
          parent_id = "2323121212"
        }
      }
      
      
      # Prod folder for team 1.
      template "folder_team1" {
        recipe_path = "{{$recipes}}/folder.hcl"
        output_path = "./live/prod"
        data = {
          parent_type  = "folder"
          display_name = "app-team"
        }
      }

Using above config gives me following error:

`Error processing module at '/Users/source/healthcare-data-protection-suite/output/live/folder/terragrunt.hcl'. How this module was found: dependency of module at ''/Users/source/healthcare-data-protection-suite/output/live/prod/folder'. Underlying error: Error reading file at path '/Users/source/healthcare-data-protection-suite/output/live/folder/terragrunt.hcl: open '/Users/source/healthcare-data-protection-suite/output/live/folder/terragrunt.hcl: no such file or directory`

Regards,
Giriraj

### PR DESCRIPTION
**Pull request by umairidris**
Fix #40 
Fix #49 
Fix #329
Partial #202 

- Move components from org/* and folder/* to top level components dir
- Support parent_type being set to folder for all components
- Add new field `add_parent_folder_dependency` to support creating new folders in a root parent folder
- Add new folder_foundation example
- Rename simple.hcl to org_foundation.hcl
- Some minor cleanup (use logging sink resource, use trimprefix func, etc)

### CODE REVIEW

File: build/gen_check.sh [hunk 1]
```diff
@@ -48,5 +48,6 @@ EOF
 
 # Run the tests
 test_gen 'go run ./cmd/policygen --config_path=examples/policygen/config.yaml --state_path examples/policygen/example.tfstate --output_path' 'examples/policygen/generated'
-test_gen 'go run ./cmd/tfengine --config_path=examples/tfengine/simple.hcl --output_path' 'examples/tfengine/generated/simple'
```
Comment (xingao267) [inline build/gen_check.sh:51]: After Martin's PR is in, can we send a PR first to just do the renaming from simple to org_foundation so we don't see a lot of deletion in this one.

File: examples/tfengine/generated/folder_foundation/live/audit/resources/main.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# - Organization IAM Audit log configs (https://cloud.google.com/logging/docs/audit),
+# - BigQuery log sink creation and configuration for short term log storage,
+# - Cloud Storage log sink creation and configuration for long term log storage,
+# - IAM permissions to grant log Auditors iam.securityReviewer role to view the logs.
+
+terraform {
+  backend "gcs" {}
+}
+
+
```
Comment (xingao267) [inline examples/tfengine/generated/folder_foundation/live/audit/resources/main.tf:25]: Add a TODO to add the folder level audit config?

File: examples/tfengine/generated/org_foundation/live/audit/resources/main.tf [hunk 3]
```diff
@@ -38,21 +40,16 @@ resource "google_organization_iam_audit_config" "config" {
   }
 }
 
-# BigQuery log sink.
-module "bigquery_log_export" {
```
Comment (xingao267) [inline examples/tfengine/generated/org_foundation/live/audit/resources/main.tf:42]: This PR is getting a bit too big. Maybe also consider do a PR that prepares adding support for folder which includes changing module to resource, etc.

File: build/gen_check.sh (hunk 1 above)
Comment (umairidris) [reply build/gen_check.sh]: Good idea, sent #332

**Comment by umairidris** [comment]
I'm going to send a new PR because a lot of these changes were moved and there are too many merge conflicts for me to resolve...

**Pull request closed without merging**